From eb78989e8493790e376ba2705e985253aa839e12 Mon Sep 17 00:00:00 2001
From: wjb <jaybob413@gmail.com>
Date: Sat, 15 Mar 2014 14:46:23 -0400
Subject: [PATCH] Add trickplay mod

---
 xbmc/cores/dvdplayer/DVDPlayer.cpp |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/xbmc/cores/dvdplayer/DVDPlayer.cpp b/xbmc/cores/dvdplayer/DVDPlayer.cpp
index cfd3dc3..cd1c01b 100644
--- a/xbmc/cores/dvdplayer/DVDPlayer.cpp
+++ b/xbmc/cores/dvdplayer/DVDPlayer.cpp
@@ -86,6 +86,9 @@
 #include "URL.h"
 #include "utils/LangCodeExpander.h"
 
+// check if in rewind mode (trickplay)
+#define RW(speed)  ((speed) < 0)
+
 using namespace std;
 using namespace PVR;
 
@@ -2324,6 +2327,14 @@ void CDVDPlayer::HandleMessages()
         // videoplayer just plays faster after the clock speed has been increased
         // 1. disable audio
         // 2. skip frames and adjust their pts or the clock
+
+        // when switching from trickplay to normal, we may not have a full set of reference frames
+        // in decoder and we may get corrupt frames out. Seeking to current time will avoid this.
+        if ( RW(speed) || RW(m_playSpeed) ||
+           ( (speed == DVD_PLAYSPEED_PAUSE || speed == DVD_PLAYSPEED_NORMAL) &&
+             (m_playSpeed != DVD_PLAYSPEED_PAUSE && m_playSpeed != DVD_PLAYSPEED_NORMAL) ) )
+          m_messenger.Put(new CDVDMsgPlayerSeek(GetTime(), (speed < 0), true, false, false, true));
+
         m_playSpeed = speed;
         m_caching = CACHESTATE_DONE;
         m_clock.SetSpeed(speed);
-- 
1.7.9.5

