--- a/amplayer/Makefile.mk	2013-05-23 14:11:43.319188503 +0200
+++ b/amplayer/Makefile.mk	2013-05-23 14:12:08.550661943 +0200
@@ -45,7 +45,6 @@ DIRS += player/system/
 
 CFLAGS+=-I${SRCTREE}/../amffmpeg -I${SRCTREE}/../amcodec/include -I${SRCTREE}/../amadec/include
 CFLAGS+=-fPIC -DENABLE_FREE_SCALE
-CFLAGS+=-mfloat-abi=softfp -mfpu=neon -mtune=cortex-a9 -march=armv7-a
 target_all=  $(TARGET)
 
 
