--- a/usbmount
+++ b/usbmount
@@ -92,10 +92,14 @@ if [ "$1" = add ]; then
     UUID=$(echo "$DEVINFO"   | sed 's/.*[[:blank:]]UUID="\([^"]*\)".*/\1/g; s/[[:blank:]]*//g;')
     USAGE=$(echo "$DEVINFO"  | sed 's/.*[[:blank:]]USAGE="\([^"]*\)".*/\1/g; s/[[:blank:]]*//g;')
 
-    if ! echo $USAGE | egrep -q "(filesystem|disklabel)"; then
-	log info "$DEVNAME does not contain a filesystem or disklabel"
-	exit 1
-    fi
+# blkid from busybox doesn't return USAGE, which causes usbmount to fail.
+# remove this check to allow processing to continue.
+# FIXME: look for an alternative to this for usb/sd mounts.
+#
+#    if ! echo $USAGE | egrep -q "(filesystem|disklabel)"; then
+#	log info "$DEVNAME does not contain a filesystem or disklabel"
+#	exit 1
+#    fi
 
     # Try to use specifications in /etc/fstab first.
     if egrep -q "^[[:blank:]]*$DEVNAME" /etc/fstab; then
